app/src/Common/Config/Settings.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Common\Config;
+
+use Exception;
+
+class Settings
+{
+    /**
+     * @return array of settings
+     */
+    public static function build()
+    {
+        // Build application settings
+
+        // Load settings from files
+        try {
+            // Directories list where config files located
+            $directories = array(
+                CONFIG_PATH,
+            );
+
+            // Locator to find files
+            $locator = new FileLocator($directories);
+
+            // Load config file into an array
+            $loader = new PHPConfigLoader($locator);
+
+            // Import all config sections
+            $settings = $loader->importAll();
+
+        } catch (Exception $ex) {
+            die($ex->getMessage());
+        }
+
+        // Load settings from ENV vars
+        $settings['params']['env'] = @getenv('APPLICATION_ENV');
+        $settings['accessToken']['secret_key'] = php_sapi_name() == 'cli-server' ? 'test-key' : @getenv('SECRET_KEY');
```

Comment:
>Is it safe?

Do not see how copy value into variable can be unsafe. It is not cookie or anything - just variable.
Well, one can say something like 'variable can be dumped into logs or whatever', but if you can read system logs - most likely you have SSH access already - no place to worry about ENV VARS

>I think, that better always getting secret key from environment.

Yes, and it is extracted from ENV VAR into local variable. It is just done in one place - not everywhere around the code - where it is required. Secret keys are part of the system settings, after all, so it seems to be reasonable to treat them the same way inside code:

1. fetch  key from ENV VAR into local settings var
2. Everywhere you need key do not use getenv or whatever, just use settings
3. In case you need to do something with key - it can be done once - inside settings.

So, I totally agree, that keys should be passed inside code via ENV VARS (and they are right now) - but I also think that keys should be passed inside code ONLY ONCE - in some centralized place - like settings reader (and not everywhere key is needed) and be used in code not directly from ENV VARS, but via settings.